src/Plugin.php

```diff
@@ -240,6 +240,48 @@ public static function wpseo_disable_adjacent_rel_links() {
     return get_option(Plugin::L10N . '_wpseo_disable_adjacent_rel_links');
   }
 
+  /**
+   * Adds the passed argument as query parameter to a set of list elements.
+   *
+   * @param string $content
+   *   The HTML filter list to perform the transformation on.
+   * @param string $arg
+   *   Use delivery time as default argument.
+   *
+   * @return \DOMDocument|null
+   */
+  public static function transformFilterLinks(string $content, $arg = 'delivery_time'): ?\DOMDocument {
+    $args = [];
+    if ($arg = $_GET[$arg] ?? '') {
+      $args = array_filter(array_map('absint', explode(',', wp_unslash($arg))));
+    }
+    // Return early if no deliver time filter is not active.
+    if (!$args) {
+      return NULL;
+    }
+
+    $doc = new \DOMDocument();
+    $charset = get_option('blog_charset');
+    $options = LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD;
+    $doc->loadHTML(mb_convert_encoding($content, 'HTML-ENTITIES', $charset), $options);
+    $xpath = new \DOMXPath($doc);
+
+    // Select all filter links and add the query parameter to it.
+    $anchors = $xpath->query('//ul/li/a');
```

Comment:
Not clear why we need to limit to lists?

```suggestion
    $anchors = $xpath->query('//a');
```